src/core/ControllerConnection.cpp

```diff
@@ -186,11 +189,10 @@ void ControllerConnection::saveSettings( QDomDocument & _doc, QDomElement & _thi
 		}
 		else
 		{
-			int id = Engine::getSong()->controllers().indexOf( m_controller );
-			if( id >= 0 )
-			{
-				_this.setAttribute( "id", id );
-			}
+			const auto& controllers = Engine::getSong()->controllers();
+			const auto it = std::find(controllers.begin(), controllers.end(), m_controller);
+			const int id = it != controllers.end() ? std::distance(controllers.begin(), it) : -1;
+			_this.setAttribute("id", id);
```

Comment:
Still not handling the `-1` case properly. I'd rather write it like this;
```suggestion
			if (it != controllers.end())
			{
						const int id = std::distance(controllers.begin(), it);
						_this.setAttribute("id", id);
			}
```

Reply:
I was thinking the original code set the ``id`` attribute, even if the controller was not found in the vector. But, after taking another look, you're right, since it was only set when it was found.

Will fix shortly.

Reply:
Oh, I put some extra indents inside the `if`. Sorry for bothering, but could you fix that as well?